Core/src/org/sleuthkit/autopsy/mainui/datamodel/AnalysisResultConfigSearchParam.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2021 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.mainui.datamodel;
+
+import java.util.Objects;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+
+/**
+ * Base class for search params for analysis results that filter by configuration.
+ */
+public class AnalysisResultConfigSearchParam extends AnalysisResultSearchParam {
```

Comment:
@gdicristofaro I think we should combine AnalysisResultConfigSearchParam and AnalysisResultSearchParam instead of extending. Basically I think AnalysisResultConfigSearchParam should become AnalysisResultSearchParam - all analysis results should have a concept of configuration, though for some analysis results it will be empty. Do you see any reason not to do this?

Reply:
Yes, AnalysisResultSearchParam will contain configuration parameter, to support all analysis results that use configuration - interesting items, hash set hits, KWS. There will be no AnalysisResultSetSearchParam class.